streams/src/main/java/org/apache/kafka/streams/state/KeyValueStore.java

```diff
@@ -51,6 +51,16 @@
     abstract public void put(K key, V value);
 
     /**
+     * Update the value associated with this key, unless a value
+     * already is associated with the key
```

Comment:
@guozhangwang The reason why I said the above is that `putIfAbsent` is a method that exists in `ConcurrentMap` and atomicity is guaranteed when it is called by multiple threads. We can write it differently (eg say that this is meant to be called from a single thread only), but I would not claim atomicity in that case (it's a bit of a meaningless statement in my opinion).